### ISSUE TITLE
[BUG] Inventory Deletion on Death

### ISSUE THREAD

**Issue by DeskDesk** [Rebirth-of-the-Night/Rebirth-Of-The-Night#524]
**Describe the bug**
Getting killed by "Magic" results in inventory being deleted, including baubles, equipment and hotbar.

When getting killed by the boss shown in the pictures everything in my inventory was deleted. Although I have not been able to test again with this boss.

I instead attempted to reproduce this bug with another player as this also happened to me when getting killed by a player.

**To Reproduce**
After a very long and annoying process we have found that.

Getting killed by someone with a Pride pendant and an onyx battleaxe with runed 3 and unbreaking 1 - unbreaking 3  will always delete your inventory, baubles, hotbar, everything.

This can potentially also happen with a greatsword and pride pendant but with varying enchantments, although runed seems to be needed on the greatsword still.
Onyx greatsword tested with is unbreaking 3, runed 3, swiftness 4, mending,  looting 3, sharpness 4

**Expected behavior**
Not to have all items in my inventory deleted and instead just respawn at spawn point.

**Screenshots**
When fighting this thing.
![image](https://user-images.githubusercontent.com/7812220/102838010-2306c380-43cb-11eb-8ae4-db62421a1f14.png)
I hit it and it hit me and I died. I could have been killed by thorns and thus killing myself with a onyx greatsword that has been confirmed to cause the bug.
![image](https://user-images.githubusercontent.com/7812220/102837954-05d1f500-43cb-11eb-93b8-c77a1c88c9b0.png)

**Info (please complete the following information):**
 - Modpack Version:  2.77.5
 - Server
 - Version the world was created on: 2.77.5

**Additional context**
It's possible something may be invoking something from the twilight forest mod that is currently disabled in 2.77.5 and causing twilight forest to attempt to read inventory data and thus deleting it since mod is disabled? IDK?
[00:51:16] [Server thread/INFO] [gamestages]: Saved 7 stages for DeskDesk.
[00:51:36] [Server thread/ERROR] [epicsiegemod]: Unable to read potion type minecraft:weakness:300:1
[00:51:57] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was killed by magic
[00:51:57] [Server thread/WARN] [twilightforest]: Dropping inventory items previously held in reserve for player DeskDesk
[00:51:57] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was slain by DeskDesk
[00:52:01] [Server thread/INFO] [gamestages]: Saved 7 stages for DeskDesk.
[00:52:46] [Server thread/INFO] [gamestages]: Saved 7 stages for DeskDesk.
[00:53:15] [Server thread/INFO] [net.minecraft.network.NetHandlerPlayServer]: DeskDesk lost connection: Disconnected

This also happened once before when I was killing witches and was accidentally killed by another player.

[03:10:00] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was killed by magic
[03:10:00] [Server thread/WARN] [twilightforest]: Dropping inventory items previously held in reserve for player DeskDesk
[03:10:01] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was slain by Pyroman852

I can confirm that with each instance of my inventory getting deleted that the following messages pop up in the logs.

[02:04:46] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was killed by magic
[02:04:46] [Server thread/WARN] [twilightforest]: Dropping inventory items previously held in reserve for player DeskDesk
[02:04:46] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was slain by Pyroman852
[02:04:47] [Server thread/INFO] [com.dynamic_view.DynView]: Mean tick: 49ms decreasing chunk view distance to: 5
[02:04:56] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was killed by magic
[02:04:56] [Server thread/WARN] [twilightforest]: Dropping inventory items previously held in reserve for player DeskDesk
[02:04:56] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was slain by Pyroman852
[02:07:11] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was killed by magic
[02:07:11] [Server thread/WARN] [twilightforest]: Dropping inventory items previously held in reserve for player DeskDesk
[02:07:11] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: DeskDesk was slain by DBHorbert

Quality of the weapon and pride pendant does not seem to matter.

**Comment by democat3457** [comment]
Oooo this is one of the juiciest bug reports I've seen. Lovely testing beforehand, thanks!

**Comment by DeskDesk** [comment]
> Oooo this is one of the juiciest bug reports I've seen. Lovely testing beforehand, thanks! Unbreaking is necessary to reproduce, I'd assume?

I have not been able to reproduce it without unbreaking yet. I may be able to do more testing this week.

**Comment by CalaMariGold** [comment]
Please test in 3.0

**Comment by CalaMariGold** [comment]
Closing because I've been killed by magic many times and inv was never deleted

**Comment by democat3457** [comment]
Well fyi, magic damage is a very general term and doesn't accurately represent what's going on under the surface of the code.

**Comment by democat3457** [comment]
Have you tested it in 3.0?

**Comment by DeskDesk** [comment]
I will test soon

**Comment by DeskDesk** [comment]
> Have you tested it in 3.0?


Yes, I did a quick test. [https://www.youtube.com/watch?v=soCZmeMtLX8](url)

Short answer is yes, it is still a bug in Beta 2.
Runed 3 looks like it can delete inv without pride amulet on, but Runed 1 may require it?
Didn't have a lot of time to test.

**Comment by CalaMariGold** [comment]
I noticed two zombie players spawned, meaning that the player died twice, which is probably what deleted the items.

**Comment by DeskDesk** [comment]
> I noticed two zombie players spawned, meaning that the player died twice, which is probably what deleted the items.

I didn't notice that in 2.7, but that doesn't mean it didn't happen.

**Comment by KaleThyself** [comment]
Bump, this issue occurs in 3.1.1a due to the wither applying "magic" damage which can cause the player to die twice. 
![Screenshot_20230831-103425](https://github.com/Rebirth-of-the-Night/Rebirth-Of-The-Night/assets/50285548/5b807b8d-d52b-4bfc-87a0-843ec91ef84e)
